### ISSUE TITLE
Authentication tutorial error: value must be an object, undefined handler

### ISSUE THREAD

**Issue by hickeyadamc** [hapijs/hapijs.com#102]
Hey!

The Authentication tutorial gives the error: Invalid server options 3000, value must be an object. 

Copy pasted from tutorial: 

```
var Bcrypt = require('bcrypt');
var Hapi = require('hapi');
var Basic = require('hapi-auth-basic');

var server = new Hapi.Server( 3000 );

var users = {
    john: {
        username: 'john',
        password: '$2a$10$iqJSHD.BGr0E2IxQwYgJmeP3NvhPrXAeLSaGCj6IR/XU5QtjVu5Tm',   // 'secret'
        name: 'John Doe',
        id: '2133d32a'
    }
};

var validate = function (username, password, callback) {
    var user = users[username];
    if (!user) {
        return callback(null, false);
    }

    Bcrypt.compare(password, user.password, function (err, isValid) {
        callback(err, isValid, { id: user.id, name: user.name });
    });
};

server.register(Basic, function (err) {
    server.auth.strategy('simple', 'basic', { validateFunc: validate });
    server.route({ method: 'GET', path: '/', config: { auth: 'simple' } });
});
```

That error can be fixed by changing:

```
var server = new Hapi.Server( );
server.connection({ port: 3000 });
```

Which results in a new error:

```
/Users/adamhickey/Prototype/node_modules/hapi/node_modules/hoek/lib/index.js:663
    throw new Error(msgs.join(' ') || 'Unknown error');
          ^
Error: Missing or undefined handler: GET /
    at Object.exports.assert (/Users/adamhickey/Prototype/node_modules/hapi/node_modules/hoek/lib/index.js:663:11)
    at new module.exports.internals.Route (/Users/adamhickey/Prototype/node_modules/hapi/lib/route.js:35:10)
    at internals.Connection._addRoute (/Users/adamhickey/Prototype/node_modules/hapi/lib/connection.js:342:17)
    at internals.Connection._route (/Users/adamhickey/Prototype/node_modules/hapi/lib/connection.js:334:18)
    at internals.Plugin._apply (/Users/adamhickey/Prototype/node_modules/hapi/lib/plugin.js:432:14)
    at internals.Plugin.route (/Users/adamhickey/Prototype/node_modules/hapi/lib/plugin.js:407:10)
    at /Users/adamhickey/Prototype/server.js:30:12
    at done (/Users/adamhickey/Prototype/node_modules/hapi/node_modules/items/lib/index.js:30:25)
    at Object.exports.register (/Users/adamhickey/Prototype/node_modules/hapi-auth-basic/lib/index.js:15:5)
    at /Users/adamhickey/Prototype/node_modules/hapi/lib/plugin.js:235:14

```

**Comment by Npednekar9** [comment]
server.register(require('inert'), (err) => { 
 
    if (err) {         throw err;     } 
 
    server.route({         
    	method: 'GET',         
    	path: '/mydemo',         
    	handler: function (request, reply) 
    	{             
    		reply.file('./public/mydemo.html');         
    	}     
    }); 
 
    server.start((err) => { 
 
        if (err) {             
        	throw err;         
        }         
        server.log('info', 'Server running at: ' + server.info.uri);     
    }); 
}); 

I am getting an error while using the inert@4 plugin in hapi@17 in the above code. Can some one please help me fix this? Following is the error I am able to see on cmd.
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): AssertionError [ERR_ASSERTION]: Invalid register options "value" must be an object
(node:14584) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by biswarupchakravarty** [comment]
@Npednekar9 

Try switching to an `async/await` structure. This is what works for me:

```javascript
async function startServer() {
  await server.register(require('inert'));
  server.route({
    method: 'GET',
    path: '/mydemo',
    handler: function(request, reply) {
      reply.file('./public/mydemo.html');
    }
  });

  await server.start();
});
startServer();
```

Check [this issue](https://github.com/hapijs/hapi/issues/3658) for further details.
TLDR; v17 moved to an async/await approach, and away from callbacks, and the documentation is in process of being updated. Alternatively, you could downgrade to v16.
